x/gmp/client/cli/tx.go

```diff
@@ -47,25 +49,69 @@ func GetCmdRelay() *cobra.Command {
 				return fmt.Errorf("destination-chain cannot be empty")
 			}
 			if args[1] == "" {
-				return fmt.Errorf("destination-address cannot be empty")
+				return fmt.Errorf("ojo-contract-address cannot be empty")
+			}
+			if args[2] == "" {
+				return fmt.Errorf("client-contract-address cannot be empty")
 			}
 			if args[3] == "" {
+				return fmt.Errorf("command-selector cannot be empty")
+			}
+			if args[4] == "" {
+				return fmt.Errorf("command-params cannot be empty")
+			}
+			if args[5] == "" {
+				return fmt.Errorf("timestamp cannot be empty")
+			}
+			if args[6] == "" {
 				return fmt.Errorf("denoms cannot be empty")
 			}
 
-			// Normalize the coin denom
-			coin, err := sdk.ParseCoinNormalized(args[2])
+			coin := sdk.Coin{}
+			// normalize the coin denom
+			if args[7] != "" {
```

Comment:
This also requires an extra keeper method like `RelayPriceWithToken` as an example since the current `RelayPrice` method only sends a `TypeGeneralMessage` by default, not a `TypeGeneralMessageWithToken`.  However since the current contract implementation is only sending regular General Messages we don't really need to implement this now.

Reply:
👍 
Yeah I think we're going to run into some fees issues and we'll have to use `TypeGeneralMessageWithToken` either way tbh. But we'll have to find that in devnet